repo/packages/K/kafka/2/config.json

```diff
@@ -0,0 +1,124 @@
+                        },
+                        "kafka": {
+                            "type": "string",
+                            "description": "Kafka distro URL",
+                            "default": "https://d1vubr0evspla.cloudfront.net/kafka_2.10-0.8.2.1.tgz"
+                        },
+                        "jar": {
+                            "type": "string",
+                            "description": "kafka-mesos*.jar URL",
+                            "default": "https://github.com/mesos/kafka/releases/download/0.9.3.0/kafka-mesos-0.9.3.0.jar"
+                        }
+                    },
+                    "required": ["cpus", "mem", "instances", "jre", "kafka", "jar"]
+                },
+
+                "zk": {
+                    "type": "string",
+                    "description": "ZK URL for Kafka",
+                    "default": "master.mesos:2181"
+                },
+                "api": {
+                    "type": "string",
+                    "description": "Scheduler API URL to bind to",
+                    "default": "http://$HOST:$PORT0"
+                },
+                "storage": {
+                    "type": "string",
+                    "description": "State storage path. See --storage option",
+                    "default": "zk:/kafka-mesos"
+                },
+                "jre": {
+                    "type": "string",
+                    "description": "See --jre option",
+                    "default": "jre-7u79-openjdk.tgz"
+                },
+
+                "framework-name": {
+                    "type": "string",
+                    "description": "Mesos Framework Name",
+                    "default": "kafka"
+                },
+                "framework-role": {
+                    "type": "string",
+                    "description": "Mesos Framework Role",
+                    "default": "*"
+                },
+
+                "user": {
+                    "type": "string",
+                    "description": "Mesos user to run tasks",
+                    "default": ""
+                },
+                "principal": {
+                    "type": "string",
+                    "description": "Mesos principal (username)",
+                    "default": ""
+                },
+                "secret": {
+                    "type": "string",
+                    "description": "Mesos secret (password)",
+                    "default": ""
+                },
+
+                "other-options": {
+                    "type": "string",
+                    "description": "Other Scheduler options if required",
+                    "default": ""
+                }
+            },
+            "required": [
+                "app",
+                "zk", "api", "storage", "jre",
+                "framework-name", "framework-role",
+                "user", "principal", "secret",
```

Comment:
required means that it is used by the marathon json in a way that isn't conditional.  If there is a condition check then it shouldn't be listed as required.  So... user, principal, secret are NOT required.